javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserClassDeclaration.java

```diff
@@ -293,14 +293,31 @@ public boolean isTypeParameter() {
 
         // We want to avoid infinite recursion in case of Object having Object as ancestor
         if (!(Object.class.getCanonicalName().equals(getQualifiedName()))) {
-            ResolvedReferenceType superclass = getSuperClass();
+            ResolvedReferenceType superclass;
+            try {
+                superclass = getSuperClass();
+            } catch (UnsolvedSymbolException e) {
+                // in case we could not resolve the super class, we may still be able to resolve (some of) the
```

Comment:
I think it could be useful to somehow track this. Callers of `getAncestors` have no way to know that some ancestors where skipped. This could lead to confusing results later on

Reply:
Perhaps we could add a flag to the method to specify if we are ok obtaining an incomplete list of ancestors. Otherwise we could return some stubs representing unresolved ancestors.

Reply:
Ok, I'll add a commit to do that. :-)

Reply:
There we go. What do you think? :-)